lib/cc/engine/analyzers/engine_config.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Engine
+    module Analyzers
+      class EngineConfig
+        attr_reader :config
+
+        def initialize(hash)
+          normalize(hash)
+        end
+
+        def language(language)
+          config.
+            fetch("languages", {}).
+            fetch(language, {})
+        end
+
+        def languages
+          config.fetch("languages", {})
+        end
+
+        private
+
+        attr_reader :hash
+
+        def normalize(hash)
+          @config = hash.fetch("config", {})
+          @config["languages"] = build_language_config(@config)
```

Comment:
Actually, it's easier just to return the new config instead of making the method mutate config.